docs/tutorials/services/navigate-with-rover-base.md

```diff
@@ -0,0 +1,646 @@
+
+### Programmatic method
+
+If you want to do this programmatically, use the service's [API method `AddWaypoint()`](/services/navigation/#addwaypoint) like the following:
+
+#### Add waypoints
+
+{{< tabs >}}
+{{% tab name="Go" %}}
+
+```go
+myNav, err := navigation.FromRobot(robot, "my_nav_service")
+
+// Create a new waypoint at the specified latitude and longitude
+location = geo.NewPoint(40.76275, -73.96)
+
+// Add your waypoint to the service's data storage
+err := myNav.AddWaypoint(context.Background(), location, nil)
+```
+
+{{% /tab %}}
+{{% tab name="Python" %}}
+
+```python
+my_nav = NavigationClient.from_robot(robot=robot, name="my_nav_service")
+
+# Create a new waypoint at the specified latitude and longitude
+location = GeoPoint(latitude=40.76275, longitude=-73.96)
+
+# Add your waypoint to the service's data storage
+await my_nav.add_waypoint(point=location)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+#### Begin navigation
+
+To start navigating, set your service to `MODE_WAYPOINT` with the service's [API method `SetMode()`](/services/navigation/#setmode):
+
+{{< tabs >}}
+{{% tab name="Go" %}}
+
+```go
+myNav, err := navigation.FromRobot(robot, "my_nav_service")
+
+// Set the Mode the service is operating in to MODE_WAYPOINT and begin navigation
+mode, err := myNav.SetMode(context.Background(), Mode.MODE_WAYPOINT, nil)
+```
+
+{{% /tab %}}
+{{% tab name="Python" %}}
+
+```python
+my_nav = NavigationClient.from_robot(robot=robot, name="my_nav_service")
+
+# Set the Mode the service is operating in to MODE_WAYPOINT and begin
+# navigation
+await my_nav.set_mode(Mode.ValueType.MODE_WAYPOINT)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Next steps: automate obstacle detection
+
+In this tutorial, you have learned how to use Navigation to navigate across waypoints.
+Now, you can make navigation even better with automated obstacle detection.
+
+First, configure a depth [camera](/components/camera) that your robot can sense how far away it is from obstacles.
+
+We configured ours as an [Intel RealSense Camera](https://www.intelrealsense.com/depth-camera-d435/), which is available as a [modular resource](/extend/modular-resources/) [in the Viam registry](https://app.viam.com/module/viam/realsense):
+
+![An example configuration for an Intel RealSense camera in the Viam app Config Builder.](/tutorials/navigate-with-rover-base/realsense-camera-config-builder.png)
+
+{{< alert title="Tip" color="tip" >}}
+
+You can alternatively use an [`ultrasonic` sensor](/components/sensor/ultrasonic) configured as a camera.
+Attribute information for an `ultrasonic` [camera](/components/camera) is the same as [for a sensor](/components/sensor/ultrasonic).
+
+{{< /alert >}}
+
+Now, if you want the robot to be able to automatically detect obstacles in front of it, [configure a Vision service segmenter](/services/vision/segmentation/).
```

Comment:
Just because you already have one now in close proximity
```suggestion
If you want the robot to be able to automatically detect obstacles in front of it, [configure a Vision service segmenter](/services/vision/segmentation/).
```

Reply:
Done